### PR TITLE
Issue-1475: Change ArtifactEntry properties date types into datetime

### PR DESCRIPTION
**Pull request by evsikov**
Fixes for strongbox/strongbox#1475.

Changed `ArtifactEntry` properties:
* `created`
* `lastUsed`
* `lastUpdated`
from type `date` to type `datetime`.

### CODE REVIEW

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:o="http://www.unbroken-dome.org/schema/liquibase-orientdb"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.2.xsd">
+
+    <changeSet id="v1.0.0.21" author="strongbox-dev@carlspring.com">
+
+        <o:changePropertyTypeDateToDatetime propertyName="created" className="ArtifactEntry"/>
```
Comment (fuss86) [inline strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml]: as commented in https://github.com/strongbox/liquibase-orientdb/pull/2, `propertyName` doesn't match your schema

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+        <o:changePropertyTypeDateToDatetime propertyName="lastUpdated" className="ArtifactEntry"/>
```
Comment (fuss86) [inline strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml]: same here

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+        <o:changePropertyTypeDateToDatetime propertyName="lastUsed" className="ArtifactEntry"/>
```
Comment (fuss86) [inline strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml]: and here as well

**Review by fuss86 — CHANGES_REQUESTED** [review]

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml (hunk 1 above)
Comment (evsikov) [reply strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml]: @fuss86 in fact all this complexity was not needed. Simply dropping meta info about 'created' property and recreating it again with a new data type was enough. So we don't need any changes into liquibase-orientdb

Comment (fuss86) [reply strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml]: Will it update previous values to date with time ?

Comment (evsikov) [reply strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml]: @fuss86, previous values don't have time information. The moment the java.util.Date hit the database with 'created' property of type date, time part of it was lost. 
By the time we apply the migration, there's no time information in orientDb.

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:o="http://www.unbroken-dome.org/schema/liquibase-orientdb"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.2.xsd">
+
```
Comment (sbespalov) [inline strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml:6]: @evsikov can you please add some comments explaining how OrientDB will behave with this ChangeSet. Something similar to your [comment](https://github.com/strongbox/strongbox/pull/1504#issuecomment-539712722).

**Comment by sbespalov** [comment]
@evsikov this new changeset looks fantastic! thank you very much for your efforts improving it!

@steve-todorov @fuss86 I've checked OrientDB statements on my local DB instance (with lot of data) and it seems works as expected, so we probably can proceed and merge it.

**Review by sbespalov — APPROVED** [review]

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml (hunk 4 above)
Comment (evsikov) [reply strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.21__ArtifactEntry.xml:6]: Done

**Review by fuss86 — APPROVED** [review]
Perfect :) LGTM

**Pull request merged**